devices/src/virtio/queue.rs

```diff
@@ -265,42 +242,14 @@ impl Queue {
 
     /// Pop the first available descriptor chain from the avail ring.
     pub fn pop<'a, 'b>(&'a mut self, mem: &'b GuestMemory) -> Option<DescriptorChain<'b>> {
-        if self.len(mem) == 0 {
+        let virtq_avail_ring =
+            VirtqAvailPtr::new(self.avail_ring, mem, self.actual_size()).unwrap();
```

Comment:
I think it would be more helpful to return `None` here instead of `unwrap`ping

Reply:
Yes, I missed this. Done.